sechub-analyzer-cli/src/main/java/com/daimler/sechub/analyzer/core/FileAnalyzer.java

```diff
@@ -0,0 +1,91 @@
+package com.daimler.sechub.analyzer.core;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.daimler.analyzer.model.Marker;
+import com.daimler.analyzer.model.MarkerPair;
+import com.daimler.analyzer.model.MarkerType;
+
+/**
+ * Searches through a file looking for SecHub markers
+ */
+public class FileAnalyzer {
+    private static final String NOSECHUB = "NOSECHUB";
```

Comment:
What If there is a text containing just a 
"NOSECHUB"  somewhere but not being a comment - e.g. a "// if this is not a real finding add NOSECHUB there"
in combination with a real finding?

For example:


```java

public void potentialEvilCommentOut(){
     // if this is not a real finding add NOSECHUB there!
      doProbablyStuffWhichCouldLeadToFalsePositive();
}
public void myEvilMethod(){
     doRealEvilStuffLikeSQLWithParamStringConcatenation();
}

public void myNotSoEvilMethod(){
     // NOSECHUB
     doingNotEvilStuffButFalsePositive();
     // NOSECHUB-END
}
```

In the case before and your current implementation potentialEvilCommentOut() method would lead to start - next start will be ignored and end from myNotSoEvilMethod() used. So code inside myEvilMethod() will be inside start-end block!

I would prefer a solution where we do explicit check for comment
"# NOSECHUB"
"// NOSECHUB"
"/* NOSECHUB\w" (I think here checking only start + whitespace is enough)
"/*     NOSECHUB\w"
"/*     NOSECHUB\w"
etc.

I think a fast approach is to check for index as done before and if there is a match use (one time compiled) regular expression patterns to remove whitespaces.

Reply:
please add real source code files like NoSechHubExample1.java to our /src/test/resources folder - so it can be loaded and you can code really. Than play around with multiple blocks inside same file and some odd situations as mentioned before and load this in your tests (like done already but with text files - so you can use dedicated editors, etc. )